cmd/snap-bootstrap/partition/sfdisk.go

```diff
@@ -281,6 +346,21 @@ func buildPartitionList(ptable *sfdiskPartitionTable, pv *gadget.LaidOutVolume)
 	return buf, toBeCreated
 }
 
+// ListCreatedPartitions returns a list of partitions created during the
+// install process.
+func ListCreatedPartitions(dl *DeviceLayout) []string {
+	parts := make([]string, 0, len(dl.partitionTable.Partitions))
+	for _, p := range dl.partitionTable.Partitions {
+		if p.Attrs != "" {
+			attrs := strings.Split(strings.TrimPrefix(p.Attrs, "GUID:"), ",")
+			if strutil.ListContains(attrs, createdPartitionAttr) && strutil.ListContains(createdPartitionGUID, p.Type) {
```

Comment:
as discussed we need to check GUID also when we create partition, we shouldn't create partition that would go undetected here